tests/rptest/tests/archival_test.py

```diff
@@ -203,14 +203,26 @@ def test_write(self):
         self.kafka_tools.produce(self.topic, 10000, 1024)
         validate(self._quick_verify, self.logger, 90)
 
-    @ignore  # https://github.com/vectorizedio/redpanda/issues/1977
     @cluster(num_nodes=3)
     def test_isolate(self):
         """Verify that our isolate/rejoin facilities actually work"""
         with firewall_blocked(self.redpanda.nodes, self._get_s3_endpoint_ip()):
             self.kafka_tools.produce(self.topic, 10000, 1024)
             time.sleep(10)  # can't busy wait here
-            self._check_bucket_is_emtpy()
+
+            # Topic manifest can be present in the bucket because topic is created before
+            # firewall is blocked. No segments or partition manifest should be present.
+            topic_manifest_id = "d0000000/meta/kafka/panda-topic/topic_manifest.json"
+            objects = self.s3_client.list_objects(self.s3_bucket_name)
+            keys = [x.Key for x in objects]
+            self.logger.debug(f"keys: {keys}")
+
+            assert len(keys) < 2, \
+                f"Bucket should be empty or contain only {topic_manifest_id}, but contains {keys}"
+
+            if len(keys) == 1:
+                assert topic_manifest_id == keys[0], \
```

Comment:
Oh, now I see here you do test for what I was wondering. The comment on the previous assertion may be misleading if it fires before this one.

Reply:
Hmm I'm thinking how to rewrite this and don't have any ideas, maybe you have?

Reply:
mm, nothing that would make it substantively better :) I think its ok